parachain/primitives/router/src/lib.rs

```diff
@@ -62,7 +87,50 @@ where
 		let network = EthereumNetworkId::get();
 
 		match payload {
-			NativeTokensPayload::Create { .. } => Vec::new().into(),
+			NativeTokensPayload::Create { token, name, symbol, decimals } => {
+				let asset_id = Self::convert_token_address(token);
+
+				let buy_execution_fee_amount = 1_000_000_000; //WeightToFee::weight_to_fee(&Weight::from_parts(600_000_000, 16_000));
+											  //= 17_476_266_491_904
+											  //=                 10
+				let buy_execution_fee = MultiAsset {
+					id: Concrete(MultiLocation::parent()),
+					fun: Fungible(buy_execution_fee_amount),
+				};
+
+				let instructions: Vec<Instruction<()>> = vec![
+					UniversalOrigin(GlobalConsensus(network)),
+					DescendOrigin(X1(Junction::AccountKey20 { network: None, key: origin.into() })),
+					WithdrawAsset(buy_execution_fee.clone().into()),
+					BuyExecution { fees: buy_execution_fee.clone(), weight_limit: Unlimited },
+					Transact {
+						origin_kind: OriginKind::SovereignAccount,
```

Comment:
If you look at how the `ForeignAssets` pallet is configured in Statemint, you'll notice that it expects the origin kind to be `OriginKind::Xcm`